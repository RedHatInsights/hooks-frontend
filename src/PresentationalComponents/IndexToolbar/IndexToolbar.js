import React, { Component } from 'react';
import {
    Toolbar,
    ToolbarGroup,
    ToolbarItem
} from '@patternfly/react-core';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export class IndexToolbar extends Component {
    render() {
        return (
            <Toolbar>
                <ToolbarGroup>
                    <ToolbarItem><Link to={ '/new' } onClick={ this.props.onClick }>New endpoint</Link></ToolbarItem>
                </ToolbarGroup>
            </Toolbar>
        );
    }
}

IndexToolbar.propTypes = {
    onClick: PropTypes.func.isRequired
};

export default IndexToolbar;
