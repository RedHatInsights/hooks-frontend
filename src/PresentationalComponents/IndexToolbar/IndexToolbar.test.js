import { IndexToolbar } from './IndexToolbar';
import { BrowserRouter as Router } from 'react-router-dom';

describe('IndexToolbar', () => {
    it('expect to render', () => {
        const wrapper = shallow(
            <Router>
                <IndexToolbar onClick={ jest.fn() } />
            </Router>
        );

        expect(wrapper.render().find('a[href="/new"]').text()).toBe('New hook');
        expect(wrapper.render()).toMatchSnapshot();
    });
});
