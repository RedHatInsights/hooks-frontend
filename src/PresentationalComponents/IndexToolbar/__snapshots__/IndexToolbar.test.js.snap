// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexToolbar expect to render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "pf-l-toolbar pf-l-toolbar",
      "style": "text-align:right",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "pf-l-toolbar__group",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "pf-l-toolbar__item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-disabled": "false",
                  "class": "pf-c-button pf-m-primary",
                  "href": "/new",
                },
                "children": Array [
                  Object {
                    "data": "New hook",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "a",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-disabled": undefined,
                  "class": undefined,
                  "href": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-disabled": undefined,
                  "class": undefined,
                  "href": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
